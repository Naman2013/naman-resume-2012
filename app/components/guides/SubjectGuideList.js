import React from 'react';
import PropTypes from 'prop-types';
import LeighTile from 'components/common/tiles/LeighTile';
import style from './SubjectGuideList.style';

const SubjectGuideList = ({ list }) => (
  <ul>
    {Array.isArray(list) && list.map(guide => <li key={`leigh-tile-${guide.title}`}><LeighTile {...guide} /></li>)}
    <style jsx>{style}</style>
  </ul>
);

SubjectGuideList.propTypes = {
  list: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    anchorText: PropTypes.string.isRequired,
    link: PropTypes.string.isRequired,
  })).isRequired,
};

export default SubjectGuideList;
