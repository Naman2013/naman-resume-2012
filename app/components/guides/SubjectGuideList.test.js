import React from 'react';
import { shallow } from 'enzyme';
import SubjectGuideList from './SubjectGuideList';

describe('SubjectGuideList', () => {
  const mockList = [{ title: 'Foo', anchorText: 'Bar', link: '#' }, { title: 'Foo', anchorText: 'Bar', link: '#' }];
  const shallowWrapper = shallow(<SubjectGuideList list={mockList} />);

  it('should render correctly', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });

  it('should paint `LeighTile` for each item provided in the list', () => {
    expect(shallowWrapper.find('LeighTile').length).toEqual(mockList.length);
  });
});
