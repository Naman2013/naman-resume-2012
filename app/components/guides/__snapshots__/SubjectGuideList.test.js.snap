// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubjectGuideList should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SubjectGuideList
    list={
      Array [
        Object {
          "anchorText": "Bar",
          "link": "#",
          "title": "Foo",
        },
        Object {
          "anchorText": "Bar",
          "link": "#",
          "title": "Foo",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <li
            className="jsx-2486111897"
          >
            <LeighTile
              anchorText="Bar"
              link="#"
              title="Foo"
            />
          </li>,
          <li
            className="jsx-2486111897"
          >
            <LeighTile
              anchorText="Bar"
              link="#"
              title="Foo"
            />
          </li>,
        ],
        <JSXStyle
          css={
            Array [
              "ul.jsx-2486111897{margin:0;padding:0;list-style-type:none;}",
              "li.jsx-2486111897{margin-bottom:20px;}",
            ]
          }
          styleId="2486111897"
        />,
      ],
      "className": "jsx-2486111897",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "leigh-tile-Foo",
        "nodeType": "host",
        "props": Object {
          "children": <LeighTile
            anchorText="Bar"
            link="#"
            title="Foo"
          />,
          "className": "jsx-2486111897",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "anchorText": "Bar",
            "link": "#",
            "title": "Foo",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "leigh-tile-Foo",
        "nodeType": "host",
        "props": Object {
          "children": <LeighTile
            anchorText="Bar"
            link="#"
            title="Foo"
          />,
          "className": "jsx-2486111897",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "anchorText": "Bar",
            "link": "#",
            "title": "Foo",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": Array [
            "ul.jsx-2486111897{margin:0;padding:0;list-style-type:none;}",
            "li.jsx-2486111897{margin-bottom:20px;}",
          ],
          "styleId": "2486111897",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <li
              className="jsx-2486111897"
            >
              <LeighTile
                anchorText="Bar"
                link="#"
                title="Foo"
              />
            </li>,
            <li
              className="jsx-2486111897"
            >
              <LeighTile
                anchorText="Bar"
                link="#"
                title="Foo"
              />
            </li>,
          ],
          <JSXStyle
            css={
              Array [
                "ul.jsx-2486111897{margin:0;padding:0;list-style-type:none;}",
                "li.jsx-2486111897{margin-bottom:20px;}",
              ]
            }
            styleId="2486111897"
          />,
        ],
        "className": "jsx-2486111897",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "leigh-tile-Foo",
          "nodeType": "host",
          "props": Object {
            "children": <LeighTile
              anchorText="Bar"
              link="#"
              title="Foo"
            />,
            "className": "jsx-2486111897",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "anchorText": "Bar",
              "link": "#",
              "title": "Foo",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "leigh-tile-Foo",
          "nodeType": "host",
          "props": Object {
            "children": <LeighTile
              anchorText="Bar"
              link="#"
              title="Foo"
            />,
            "className": "jsx-2486111897",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "anchorText": "Bar",
              "link": "#",
              "title": "Foo",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": Array [
              "ul.jsx-2486111897{margin:0;padding:0;list-style-type:none;}",
              "li.jsx-2486111897{margin-bottom:20px;}",
            ],
            "styleId": "2486111897",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
